src/utils/hooks/useListApplications.tsx

```diff
@@ -56,8 +57,16 @@ const useListApplications = (
     fetchPolicy: 'network-only',
   })
 
+  const [getListCount, { loading: loadingCount, data: countData }] =
+    useGetFilteredApplicationCountLazyQuery({
+      fetchPolicy: 'network-only',
+    })
+
   // Ensures that query doesn't request a page beyond the available total
   useEffect(() => {
+    if (applicationCount == 'loading') {
+      return
```

Comment:
I wonder if pagination is now broken and query will request beyond pagination ? (if count is only figured out once applicationList loads ?). I wonder how to replicate this use case ?